python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -197,17 +198,17 @@ def wan_local_path_group_names(self: SharedUtilsProtocol) -> list:
         return list(self.wan_local_path_groups.keys())
 
     @cached_property
-    def wan_ha_peer_path_groups(self: SharedUtilsProtocol) -> list:
+    def wan_ha_peer_path_groups(self: SharedUtilsProtocol) -> EosDesignsFactsProtocol.WanPathGroups:
         """List of WAN HA peer path-groups coming from facts."""
-        if not self.is_wan_router or not self.wan_ha:
-            return []
-        peer_facts = self.get_peer_facts(self.wan_ha_peer, required=True)
-        return get(peer_facts, "wan_path_groups", required=True)
+        if not self.is_wan_router or not self.wan_ha or not self.wan_ha_peer:
```

Comment:
we can simplify further like this because `self.wan_ha_peer` will be None if `self.wan_ha` is not True
```suggestion
        if not self.is_wan_router or not self.wan_ha_peer:
```

Reply:
done